src/client/Client.h

```diff
@@ -982,6 +1067,31 @@ class Client : public Dispatcher, public md_config_obs_t {
 
   client_t whoami;
 
+  enum _state {
+    CLIENT_NEW,
+    CLIENT_INITIALIZING,
+    CLIENT_INITIALIZED,
+    CLIENT_MOUNTING,
+    CLIENT_MOUNTED,
+    CLIENT_UNMOUNTING,
+    CLIENT_UNMOUNTED = CLIENT_INITIALIZED,
+  };
```

Comment:
This needs some documentation that explains what distinguishes the different states. Building a state machine that has poorly documented states will lead to bitrot as people subtly interpret them in different ways. A flowchart that shows how the client states change might be warranted.

Reply:
Sure, I will document it.